index.js

```diff
@@ -0,0 +1,158 @@
+import * as core from "@actions/core";
+import { LinearClient } from "@linear/sdk";
+
+const ESCAPE = {
+  ">": "&gt;",
+  "<": "&lt;",
+  "&": "&amp;",
+  "\r\n": " ",
+  "\n": " ",
+  "\r": " ",
+};
+const ESPACE_REGEX = new RegExp(Object.keys(ESCAPE).join("|"), "gi");
+
+const getBugs = async (linearClient, stateIds, label) => {
+  const issues = await linearClient.issues({
+    filter: {
+      labels: {
+        and: [{ name: { eq: "Bug" } }, label ? { name: { eq: label } } : {}],
+      },
+      state: { id: { in: stateIds } },
+      project: { null: true },
+    },
+  });
+
+  return removeChildIssues(issues);
+};
+
+const getChores = async (linearClient, stateIds, label) => {
+  const issues = await linearClient.issues({
+    filter: {
+      labels: {
+        and: [{ name: { eq: "Chore" } }, label ? { name: { eq: label } } : {}],
+      },
+      state: { id: { in: stateIds } },
+      project: { null: true },
+    },
+  });
+
+  return removeChildIssues(issues);
+};
+
+const getFastTracks = async (linearClient, stateIds, label) => {
+  const issues = await linearClient.issues({
+    filter: {
+      labels: {
+        and: [
+          { every: { name: { nin: ["Bug", "Chore"] } } },
+          label ? { name: { eq: label } } : {},
+        ],
+      },
+      state: { id: { in: stateIds } },
+      project: { null: true },
+    },
+  });
+
+  return removeChildIssues(issues);
+};
+
+const getProjects = async (linearClient, stateIds, label) => {
+  const issues = await linearClient.issues({
+    filter: {
+      project: { null: false },
+      state: { id: { in: stateIds } },
+      labels: label ? { name: { eq: label } } : {},
+    },
+  });
+  const projects = {};
+
+  for (let i = 0; i < issues.nodes.length; i++) {
+    const issue = issues.nodes[i];
+    if (!(issue._project.id in projects)) {
+      projects[issue._project.id] = await issue.project;
+    }
+  }
+
+  return Object.values(projects).sort(
+    (first, second) => second.progress - first.progress
+  );
+};
+
+const removeChildIssues = (issues) => {
+  issues.nodes = issues.nodes.filter((issue) => issue._parent === undefined);
+  return issues;
+};
+
+const formatIssues = (issues) =>
+  issues.nodes
+    .map(({ title, url }) => `- <${url}|${escapeText(title)}>`)
+    .join("\n") || "_No tickets_";
+
+const escapeText = (text) =>
+  text.replace(ESPACE_REGEX, (match) => ESCAPE[match]);
+
+const formatProjects = (projects) =>
+  projects
+    .map(({ name, url, progress, targetDate }) => {
+      name = `<${url}|${escapeText(name)}>`;
+      progress = formatProgress(progress);
+      targetDate = formatTargetDate(targetDate);
+      return `- ${name}, Progress: ${progress}, Target date: ${targetDate}`;
```

Comment:
Just to make it clearer"
```suggestion
      return `- ${name}, Progress: ${progress}, Target stage release date: ${targetDate}`;
```

Reply:
Applied.